### PR TITLE
ci: Automatically rebuild taxonomies

### PR DESCRIPTION
**Pull request by hangy**
**Description:** If a push to `master` (ie. a merged pull request) updates `taxonomies/*.txt*`, try to run `refresh_taxonomies.js` to check if the taxonomies changed content-wise from the previous state, without the `.sto` files being updated in the same push.

**Related issues and discussion:** N/A

### CODE REVIEW

**Comment by hangy** [comment]
This would avoid us having to manually update the taxonomies by automatically creating a pull request with regenerated `.sto` files.

We may want to move the `.sto` files to Git LFS before activating this, because high frequency updates of large(r) binary files is not very good when using git.

**Comment by stephanegigandet** [comment]
> This would avoid us having to manually update the taxonomies by automatically creating a pull request with regenerated `.sto` files.

That would be very useful.

**Comment by VaiTon** [comment]
@hangy is this ready for review? It would be very useful.

**Comment by hangy** [comment]
> > This would avoid us having to manually update the taxonomies by automatically creating a pull request with regenerated `.sto` files.
> 
> That would be very useful.

> @hangy is this ready for review? It would be very useful.

What do you think about a potential issue with regularly changing binary files (possibly polluting this git repository too much)?

> We may want to move the `.sto` files to Git LFS before activating this, because high frequency updates of large(r) binary files is not very good when using git.

**Comment by VaiTon** [comment]
In my opinion `.sto` files (or if we adopt `.json`) should be built only on the production/dev server. There is no reason to keep them in git if we can just run a script and rebuild them before deployment.

@hangy @stephanegigandet

**Comment by hangy** [comment]
> In my opinion `.sto` files (or if we adopt `.json`) should be built only on the production/dev server. There is no reason to keep them in git if we can just run a script and rebuild them before deployment.

I generally agree. One problem stopping us from that at this moment, is that the compiled taxonomies are required for unit tests, and the unit tests do not (yet) use static, unchanging taxonomies.

**Comment by stephanegigandet** [comment]
> unit tests do not (yet) use static, unchanging taxonomies.

Some of those unit tests are actually testing the taxonomy changes themselves, to make sure there aren't unintended changes in the taxonomies that affect many products. That's especially true for ingredients and additives.

**Comment by VaiTon** [comment]
> 
> 
> > In my opinion `.sto` files (or if we adopt `.json`) should be built only on the production/dev server. There is no reason to keep them in git if we can just run a script and rebuild them before deployment.
> 
> I generally agree. One problem stopping us from that at this moment, is that the compiled taxonomies are required for unit tests, and the unit tests do not (yet) use static, unchanging taxonomies.

We could build them on unit testing, as we install npm modules. It would mean more testing time but if we optimize the build process it shouldn't be so much.

**Comment by teolemon** [comment]
Is it still WIP ? It looks harmless if we merge @hangy ? @stephanegigandet

**Comment by hangy** [comment]
> Is it still WIP ? It looks harmless if we merge @hangy ? @stephanegigandet 

Not necessarily harmless. Every build causes several MiBs of bloat, because of the binary format of the taxonomies. I'm inclined to close this and only actually do this if the taxonomies were stored elsewhere …

**Comment by teolemon** [comment]
@hangy @stephanegigandet to solve the issue, we could only trigger it on a branch called `rebuild-taxonomies`, that we would merge master into, only on a schedule (each week, each month).

File: .github/workflows/update_taxonomies.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+name: Update Taxonomies
+
+on:
+  push:
+    branches:
+      - master
```
Comment (teolemon) [inline .github/workflows/update_taxonomies.yml:6]:
```suggestion
      - rebuild-taxonomies
```

Comment (teolemon) [reply .github/workflows/update_taxonomies.yml:6]: @stephanegigandet @hangy

**Pull request closed without merging**